k4FWCore/src/PodioDataSvc.cpp

```diff
@@ -109,10 +126,16 @@ StatusCode PodioDataSvc::i_setRoot(std::string root_path, DataObject* pRootObj)
 }
 
 void PodioDataSvc::endOfRead() {
+  m_eventNum++;
+
+  if (!m_bounds_check_needed) {
+    return;
+  }
+
   StatusCode sc;
-  if (m_eventMax != -1) {
-    if (m_eventNum++ >= m_eventMax - 1) {  // we start counting at 0 thus the -1.
-      info() << "Reached end of file with event " << m_eventMax << endmsg;
+  if (m_availableEventMax != -1) {
+    if (m_eventNum >= m_availableEventMax - 1) {  // we start counting at 0 thus the -1.
```

Comment:
```suggestion
    if (m_availableEventMax != -1 && m_eventNum + 1>= m_availableEventMax) {  // this will process the current event thus the +1.
```
(plus removing a } that I can't suggest)
I think this comment is better since it's not very intuitive because what happens is that the current event gets processed so you have to check if the next one will be beyond bounds

Reply:
Well it is correct, the current event that `m_eventNum` is pointing to will be processed unconditionally so that's why we have to check for the next event after the one that `m_eventNum` (maybe this is not explicit enough) is pointing to independently of the value that it has right now. That's why I feel the +1 is easier to understand than the -1.

Reply:
> Well it is correct, the current event that `m_eventNum` is pointing to will be processed unconditionally

No?